File: markdowner-filter/src/test/java/org/jesperancinha/parser/markdowner/filter/GradleFilterTest.java
package org.jesperancinha.parser.markdowner.filter;

import org.junit.jupiter.api.Test;

import java.nio.file.Path;

import static org.assertj.core.api.Assertions.assertThat;

public class GradleFilterTest {

    @Test
    public void testGradleFilter() {
        final var resource1 = getClass().getResource("/directory2NoReadme/project4Gradle/gradle.build");
        assertThat(resource1).isNotNull();
        final var resource = Path.of(resource1.getPath());
        final var npmFilter = new GradleFilter();

        final var result = npmFilter.test(resource);

        assertThat(result).isTrue();
        assertThat(npmFilter.lastProjectName()).isEqualTo("project4Gradle");
    }

    @Test
    public void testGradleFilterFail() {
        final var resource1 = getClass().getResource("/directory2NoReadme/project3MavenAndNPM/package.json");
        assertThat(resource1).isNotNull();
        final var resource = Path.of(resource1.getPath());
        final var mavenFilter = new GradleFilter();

        final var result = mavenFilter.test(resource);

        assertThat(result).isFalse();
    }
}

File: markdowner-filter/src/test/kotlin/org/jesperancinha/parser/markdowner/filter/GradleFilterTest.kt
package org.jesperancinha.parser.markdowner.filter

import io.kotest.matchers.booleans.shouldBeFalse
import io.kotest.matchers.booleans.shouldBeTrue
import io.kotest.matchers.nulls.shouldNotBeNull
import io.kotest.matchers.shouldBe
import org.junit.jupiter.api.Test
import java.nio.file.Path

class GradleFilterTest {
    @Test
    fun testGradleFilter() {
        val gradleBuild = javaClass.getResource("/directory2NoReadme/project4Gradle/gradle.build")
        gradleBuild.shouldNotBeNull()
        val resource = Path.of(gradleBuild.path)
        val npmFilter = GradleFilter()
        val result = npmFilter.test(resource)
        result.shouldBeTrue()
        npmFilter.lastProjectName() shouldBe "project4Gradle"
    }

    @Test
    fun testGradleFilterFail() {
        val packageJson = javaClass.getResource("/directory2NoReadme/project3MavenAndNPM/package.json")
        packageJson.shouldNotBeNull()
        val resource = Path.of(packageJson.path)
        val mavenFilter = GradleFilter()
        val result = mavenFilter.test(resource)
        result.shouldBeFalse()
    }
}